src/js/core/game_state.ts

```diff
@@ -180,39 +169,43 @@ export class GameState {
     /**
      * Should return whether to fade in the game state. This will then apply the right css classes
      * for the fadein.
-     * @returns {boolean}
      */
-    getHasFadeIn() {
+    getHasFadeIn(): boolean {
         return true;
     }
 
     /**
      * Should return whether to fade out the game state. This will then apply the right css classes
      * for the fadeout and wait the delay before moving states
-     * @returns {boolean}
      */
-    getHasFadeOut() {
+    getHasFadeOut(): boolean {
         return true;
     }
 
     /**
      * Returns if this state should get paused if it does not have focus
-     * @returns {boolean} true to pause the updating of the game
+     * @returns true to pause the updating of the game
      */
-    getPauseOnFocusLost() {
+    getPauseOnFocusLost(): boolean {
         return true;
     }
 
     /**
      * Should return the html code of the state.
-     * @returns {string}
-     * @abstract
+     * @deprecated use {@link getContentLayout} instead
      */
-    getInnerHTML() {
-        abstract;
+    getInnerHTML(): string {
         return "";
     }
 
+    /**
+     * Should return the element(s) to be displayed in the state.
+     * If null, {@link getInnerHTML} will be used instead.
+     */
+    protected getContentLayout(): Node {
+        return null;
```

Comment:
this could, by default, call `getInnerHTML()` itself and create an element to return instead of doing such in `internalGetWrappedContent()`; maybe it'd be cleaner

Reply:
Done, check if that's what you meant: https://github.com/tobspr-games/shapez-community-edition/pull/39/commits/8955e37c78bad0b93ace6f5cad4189fe777ca412